js_modules/dagster-ui/packages/ui-core/src/partitions/__tests__/OrdinalOrSingleRangePartitionSelector.test.tsx

```diff
@@ -0,0 +1,174 @@
+import {render, waitFor} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import * as React from 'react';
+
+import {
+  ONE_DIMENSIONAL_ASSET,
+  TWO_DIMENSIONAL_ASSET_BOTH_STATIC,
+} from '../../assets/__fixtures__/PartitionHealth.fixtures';
+import {PartitionHealthQuery} from '../../assets/types/usePartitionHealthData.types';
+import {
+  PartitionDimensionSelection,
+  buildPartitionHealthData,
+} from '../../assets/usePartitionHealthData';
+import {mockViewportClientRect, restoreViewportClientRect} from '../../testing/mocking';
+import {OrdinalOrSingleRangePartitionSelector} from '../OrdinalOrSingleRangePartitionSelector';
+
+const Wrapper = ({
+  queryResult,
+  initialSelection,
+}: {
+  queryResult: PartitionHealthQuery;
+  initialSelection?: PartitionDimensionSelection | null;
+}) => {
+  const assetHealth = buildPartitionHealthData(queryResult, {path: ['asset']});
+  const [selection, setSelection] = React.useState<PartitionDimensionSelection | null | undefined>(
+    initialSelection,
+  );
+  return (
+    <>
+      <OrdinalOrSingleRangePartitionSelector
+        dimension={assetHealth.dimensions[0]!}
+        health={assetHealth}
+        selection={selection}
+        setSelection={setSelection}
+      />
+      <div data-testid="selection">
+        {selection
+          ? JSON.stringify({
+              selectedKeys: selection.selectedKeys,
+              selectedRanges: selection.selectedRanges,
+            })
+          : selection === null
+          ? 'null'
+          : selection === undefined
+          ? 'undefined'
+          : ''}
+      </div>
+    </>
+  );
+};
+describe('OrdinalOrSingleRangePartitionSelector', () => {
+  beforeAll(() => {
+    mockViewportClientRect();
+  });
+  afterAll(() => {
+    restoreViewportClientRect();
+  });
+
+  it('should populate with an existing value (selectedKeys)', async () => {
+    const {getByText, getByTitle} = render(
+      <Wrapper
+        queryResult={TWO_DIMENSIONAL_ASSET_BOTH_STATIC}
+        initialSelection={{dimension: {} as any, selectedKeys: ['CA', 'MN'], selectedRanges: []}}
+      />,
+    );
+    expect(getByText('Single').parentElement).toHaveAttribute('aria-selected', 'true');
+    expect(getByTitle('CA')).toBeVisible(); // tags are in the element
+    expect(getByTitle('MN')).toBeVisible();
+  });
+
+  it('should populate with an existing value (null)', async () => {
+    const {getByText} = render(
+      <Wrapper queryResult={TWO_DIMENSIONAL_ASSET_BOTH_STATIC} initialSelection={null} />,
+    );
+    expect(getByText('All').parentElement).toHaveAttribute('aria-selected', 'true');
```

Comment:
Not a big fan of using `parentElement` in a test since it makes the test tied to the exact dom structure which is more brittle

Reply:
Hmm yeah, I couldn't find a better way to do this one -- the text is inside a span inside the button :( It's sort of odd this library has such a limited set of selectors, ideally using a querySelector would fix this!

Reply:
Edit ahh I see, I think I can forgo these find methods and write this instead:

```expect(baseElement.querySelector('[aria-selected=true]')?.textContent).toEqual('All')```

That does seem better